packages/l/lief/xmake.lua

```diff
@@ -85,7 +91,7 @@ package("lief")
                 table.insert(configs, "-DLIEF_" .. name:upper() .. "=" .. (enabled and "ON" or "OFF"))
             end
         end
-        import("package.tools.cmake").install(package, configs)
+        import("package.tools.cmake").install(package, configs, {packagedeps = {"tl_expected", "utfcpp", "mbedtls", "tcb-span", "frozen", "nlohmann_json", "spdlog", "fmt"}})
```

Comment:
In TLDR. ```xmake l scripts/test.lua -vD -k shared -a x64 --shallow --runtimes=MT lief```
Goes to fallback for vcpkg instead of xmake...
```checking for spdlog ... vcpkg::spdlog 1.15.0```
This would produce issue like
```
CMake Error at cmake/LIEFDependencies.cmake:171 (find_package):
  By not providing "Findspdlog.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "spdlog", but
  CMake did not find one.

  Could not find a package configuration file provided by "spdlog" with any
  of the following names:

    spdlogConfig.cmake
    spdlog-config.cmake

  Add the installation prefix of "spdlog" to CMAKE_PREFIX_PATH or set
  "spdlog_DIR" to a directory containing one of the above files.  If "spdlog"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  CMakeLists.txt:89 (include)
  ```

Reply:
Try `system = false` for spdlog?

Reply:
How would I bind system = false for ```checking for fmt ... vcpkg::fmt 11.0.2```

Reply:
```diff
PS D:\xm\xmake-repo> git diff
diff --git a/packages/l/lief/xmake.lua b/packages/l/lief/xmake.lua
index c120e8de..ee904580 100644
--- a/packages/l/lief/xmake.lua
+++ b/packages/l/lief/xmake.lua
@@ -32,9 +32,9 @@ package("lief")
     end

     add_deps("cmake")
-    add_deps("spdlog", {configs = {fmt_external = false, noexcept = true, header_only = false, fmt_external_ho = true}})
-    add_deps("nlohmann_json", {configs = {cmake = true}})
-    add_deps("tl_expected", "utfcpp", "mbedtls <3.6.0", "tcb-span", "frozen")
+    add_deps("spdlog", {configs = {fmt_external = false, noexcept = true, header_only = false, fmt_external_ho = true}, system = false})
+    add_deps("nlohmann_json", {configs = {cmake = true}, system = false})
+    add_deps("tl_expected", "utfcpp", "mbedtls <3.6.0", "tcb-span", "frozen", {system = false})

     if on_check then
         on_check(function (package)
diff --git a/packages/s/spdlog/xmake.lua b/packages/s/spdlog/xmake.lua
index 8ead45b4..a954fe54 100644
--- a/packages/s/spdlog/xmake.lua
+++ b/packages/s/spdlog/xmake.lua
@@ -52,7 +52,7 @@ package("spdlog")
             package:add("defines", "SPDLOG_USE_STD_FORMAT")
         elseif package:config("fmt_external") or package:config("fmt_external_ho") then
             package:add("defines", "SPDLOG_FMT_EXTERNAL")
-            package:add("deps", "fmt", {configs = {header_only = package:config("header_only")}})
+            package:add("deps", "fmt", {configs = {header_only = package:config("header_only")}, system = false})
         end
         if not package:config("header_only") and package:config("fmt_external_ho") then
             package:add("defines", "FMT_HEADER_ONLY=1")
```
Works.